### PR TITLE
Feat: Adds `EmptyState` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

`fs/components`
- [x] Moves `EmptyState` component
- [x] Adds `EmptyState ` doc page

`fs/ui`
- [x] Adds and reviews `EmptyState` styles

`fs/core`
- [x] Review `EmptyGallery` and `EmptyCart` structure (replace)
- [x] Delete  old `EmptyState`'s  related files (storybook and style)
- [x] Replaces temp Loading pages layout

## How to test it?

1. Check the `EmptyState` component using the doc preview link:
 - [EmptyState](https://faststore-site-git-feat-emtpy-state-fs-977-faststore.vercel.app/components/organisms/empty-state)

2. run `yarn dev` on `faststore/core` application and check the `EmptyState` occurrences on:
   a. Open the `CartSidebar` when the cart has no items. 
   b. Search for `Gallery` using the search input.
   c. Try the `Not Found: 404` page, ex: try http://localhost:3000/echo-smart-speaker-99988214/p
   
image

   

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/48

### CODE REVIEW

File: packages/components/src/organisms/EmptyState/EmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   */
+  title?: string
+  /**
+   * A React component that will be rendered as an icon on the Title.
+   */
+  titleIcon?: ReactNode
+  /**
+   * Specifies the component border variant.
+   */
+  variant?: Variant
```
Comment (ArthurTriis1) [inline packages/components/src/organisms/EmptyState/EmptyState.tsx:22]: I could not be able to identify the differences between the default and rounded variants, is this right?

Screenshot 2023-04-17 at 10 44 46

**Review by ArthurTriis1 — APPROVED** [review]

File: packages/components/src/organisms/EmptyState/EmptyState.tsx (hunk 1 above)
Comment (hellofanny) [reply packages/components/src/organisms/EmptyState/EmptyState.tsx:22]: There is a slight difference in the borders 👁️ 

image

File: packages/components/src/organisms/EmptyState/EmptyState.stories.mdx
Comment (lucasfp13) [inline packages/components/src/organisms/EmptyState/EmptyState.stories.mdx:1]: Do we need to keep this?

**Review by lucasfp13 — APPROVED** [review]
LGTM! 💯

File: packages/components/src/organisms/EmptyState/EmptyState.stories.mdx
Comment (hellofanny) [reply packages/components/src/organisms/EmptyState/EmptyState.stories.mdx:1]: Nops! I'll remove it :)

File: packages/core/src/pages/login.tsx [hunk 2]
```diff
@@ -12,7 +13,9 @@ function Page() {
     <>
       <NextSeo noindex nofollow />
 
-      <div>loading...</div>
+      <UIEmptyState title="Loading">
+        <UILoader />
+      </UIEmptyState>
```
Comment (renatamottam) [inline packages/core/src/pages/login.tsx:18]: Fan, do you think we could change this bg to white? Seems a little bit odd with the footer (same thing with the 404 page)

image

File: packages/core/src/components/sections/ProductGallery/EmptyGallery.tsx [hunk 3]
```diff
@@ -1,37 +1,44 @@
...
-    <EmptyState variant="rounded">
-      <header data-fs-empty-state-title>
-        <Icon name="CircleWavyWarning" width={56} height={56} weight="thin" />
-
-        <p>Nothing matches with your search</p>
-      </header>
-
-      <LinkButton
-        data-fs-empty-state-link
+    <UIEmptyState
```
Comment (renatamottam) [inline packages/core/src/components/sections/ProductGallery/EmptyGallery.tsx:9]: Can we fix this extra space between the `breadcrumb` and the `empty state`?

image

File: packages/core/src/pages/login.tsx (hunk 2 above)
Comment (hellofanny) [reply packages/core/src/pages/login.tsx:18]: agreed! Lemme change it!

File: packages/core/src/components/sections/ProductGallery/EmptyGallery.tsx (hunk 3 above)
Comment (hellofanny) [reply packages/core/src/components/sections/ProductGallery/EmptyGallery.tsx:9]: Fixed [here](https://github.com/vtex/faststore/pull/1716/commits/907bfdef05db595b3954dbe95b8999226ac2fece)

image

**Pull request merged**